mayastor/tests/lock_lba_range.rs

```diff
@@ -0,0 +1,269 @@
+        let ctx = get_shareable_ctx(1, 10);
+        let ctx_clone = Arc::clone(&ctx);
+        let ch = get_shareable_nexus_channel();
+        let ch_clone = Arc::clone(&ch);
+        reactor.send_future(async move {
+            lock_range(
+                &mut ctx_clone.lock().unwrap(),
+                &ch_clone.lock().unwrap(),
+            )
+            .await;
+            let _ = s.send(());
+        });
+        reactor_poll!(r);
+
+        // Second lock
+        let (lock_sender, lock_receiver) = unbounded::<()>();
+        let ctx1 = get_shareable_ctx(1, 5);
+        let ctx1_clone = Arc::clone(&ctx1);
+        let ch1 = get_shareable_nexus_channel();
+        let ch_clone1 = Arc::clone(&ch1);
+        reactor.send_future(async move {
+            lock_range(
+                &mut ctx1_clone.lock().unwrap(),
+                &ch_clone1.lock().unwrap(),
+            )
+            .await;
+            trace!("Second lock succeeded");
+            let _ = lock_sender.send(());
+        });
+        reactor_poll!(100);
+
+        // First lock is held, second lock shouldn't succeed
+        assert!(lock_receiver.try_recv().is_err());
+
+        // First unlock
+        let (s, r) = unbounded::<()>();
+        let ctx_clone = Arc::clone(&ctx);
+        let ch_clone = Arc::clone(&ch);
+        reactor.send_future(async move {
+            unlock_range(
+                &mut ctx_clone.lock().unwrap(),
+                &ch_clone.lock().unwrap(),
+            )
+            .await;
+            trace!("Unlock first lock");
+            let _ = s.send(());
+        });
+        reactor_poll!(r);
+
+        // First lock released, second lock should succeed
+        assert!(lock_receiver.try_recv().is_ok());
+
+        // Second unlock
+        let (s, r) = unbounded::<()>();
+        let ctx1_clone = Arc::clone(&ctx1);
+        let ch_clone1 = Arc::clone(&ch1);
+        reactor.send_future(async move {
+            unlock_range(
+                &mut ctx1_clone.lock().unwrap(),
+                &ch_clone1.lock().unwrap(),
+            )
+            .await;
+            trace!("Unlock second lock");
+            let _ = s.send(());
+        });
+        reactor_poll!(r);
+    }
+    test_fini();
+}
+
+#[test]
+// Test locking a block range and then issuing a front-end I/O to an overlapping
+// range.
+// TODO: Add additional test for issuing front-end I/O then taking a lock
+fn lock_then_fe_io() {
+    test_ini();
+    {
+        let reactor = Reactors::current();
+
+        // Lock range
+        let (s, r) = unbounded::<()>();
+        let ctx = get_shareable_ctx(1, 10);
+        let ctx_clone = Arc::clone(&ctx);
+        let ch = get_shareable_nexus_channel();
+        let ch_clone = Arc::clone(&ch);
+
+        reactor.send_future(async move {
+            lock_range(
+                &mut ctx_clone.lock().unwrap(),
+                &ch_clone.lock().unwrap(),
+            )
+            .await;
+            trace!("Lock succeeded");
```

Comment:
if you unwrap() the lock_range, the trace is not needed :)

Reply:
Done